### ISSUE TITLE
Need more tests for faculty and DCC views

### ISSUE THREAD

**Issue by s-gv** [s-gv/bheemboy#69]

**Comment by rajeshsundaresan** [comment]
Sagar,

What issues do you see? A generic action item like "Need more tests ..." 
is not of much value. Instead identify specific problems and devise 
tests to bring them out.

Rajesh

**Comment by s-gv** [comment]
We do need generic tests; tests that check the functioning of the website (even things that we know are working properly). This is needed because we don't want future code changes to accidentally break existing features that are functioning fine. I have added some tests for the student, profile, and FAQ views. Similar tests are needed for faculty and DCC views.

Tests to check if the server rejects bad requests are also needed. When I added tests for the student views, I discovered that students could delete courses even after grading is done (now fixed). Likewise, issues maybe uncovered by the tests. I think there is at least one major bug in the instructor view.
